src/app/core/_components/forms/custom-forms/superhashlist/new-superhashlist/new-superhashlist.component.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnDestroy,
+  OnInit
+} from '@angular/core';
+import {
+  AbstractControl,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators
+} from '@angular/forms';
+import { Router } from '@angular/router';
+
+import { UnsubscribeService } from 'src/app/core/_services/unsubscribe.service';
+import { AutoTitleService } from 'src/app/core/_services/shared/autotitle.service';
+import { AlertService } from 'src/app/core/_services/shared/alert.service';
+import { ListResponseWrapper } from 'src/app/core/_models/response.model';
+import { environment } from '../../../../../../../environments/environment';
+import { GlobalService } from 'src/app/core/_services/main.service';
+import { Hashlist } from 'src/app/core/_models/hashlist.model';
+import { extractIds } from '../../../../../../shared/utils/forms';
+import { SelectField } from 'src/app/core/_models/input.model';
+import { SERV } from '../../../../../_services/main.config';
+
+/**
+ * Represents the NewSuperhashlistComponent responsible for creating a new SuperHashlist.
+ */
+@Component({
+  selector: 'app-new-superhashlist',
+  templateUrl: './new-superhashlist.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class NewSuperhashlistComponent implements OnInit, OnDestroy {
+  /** Flag indicating whether data is still loading. */
+  isLoading = true;
+
+  /** Form group for the new SuperHashlist. */
+  form: FormGroup;
+
+  /** Maximum results for API requests. */
+  private maxResults = environment.config.prodApiMaxResults;
+
+  /** List of hashlists. */
+  selectHashlists: any;
+
+  // Util functions
+  /** Utility function for extracting IDs from a list of items. */
+  extractIds = extractIds;
+
+  /**
+   * Constructor of the NewSuperhashlistComponent.
+   *
+   * @param unsubscribeService - The service responsible for managing subscriptions.
+   * @param changeDetectorRef - Reference to the change detector to manually trigger change detection.
+   * @param titleService - Service for managing the title of the page.
+   * @param alert - Service for displaying alerts.
+   * @param globalService - Service for making global API requests.
+   * @param formBuilder - FormBuilder service for creating reactive forms.
+   * @param router - Angular Router service for navigation.
+   */
+  constructor(
+    private unsubscribeService: UnsubscribeService,
+    private changeDetectorRef: ChangeDetectorRef,
+    private titleService: AutoTitleService,
+    private alert: AlertService,
+    private globalService: GlobalService,
+    private formBuilder: FormBuilder,
+    private router: Router
+  ) {
+    this.buildForm();
+    titleService.set(['New SuperHashlist']);
+  }
+
+  @Input()
```

Comment:
From the styleguide:
Consider placing @[Input](https://angular.io/api/core/Input)() or @[Output](https://angular.io/api/core/Output)() on the same line as the property it decorates.